Makefile

```
@@ -106,6 +113,42 @@ service-watch:
+# `haven` (built from cmd/haven) gives every worktree's services stable
+# hostnames — app|api|gateway|nlp.<slug>.langwatch.localhost — so multiple
+# worktrees never fight over ports. `pnpm dev` routes through it automatically
+# once the portless proxy is installed. Dashboard: https://langwatch.localhost
+
+HAVEN_PKG = ./cmd/haven
+HAVEN = $$(command -v haven || echo "go run $(HAVEN_PKG)")
+
+# One-time machine setup: install the portless proxy (persistent), trust its CA,
+# and build the haven binary. `.localhost` resolves to loopback natively, so
+# there is no /etc/hosts or DNS step.
+portless-setup:
```

checkmake (0.3.2)

[warning] Target body for "portless-setup" exceeds allowed length of 5 lines (6).

(maxbodylength)